cmd/cli/trafficpolicy_check.go

```diff
@@ -0,0 +1,205 @@
+		Long:  trafficPolicyCheckDescription,
+		Args:  cobra.ExactArgs(2),
+		RunE: func(_ *cobra.Command, args []string) error {
+			trafficPolicyCheckCmd.sourcePod = args[0]
+			trafficPolicyCheckCmd.destinationPod = args[1]
+
+			config, err := settings.RESTClientGetter().ToRESTConfig()
+			if err != nil {
+				return errors.Errorf("Error fetching kubeconfig: %s", err)
+			}
+
+			clientset, err := kubernetes.NewForConfig(config)
+			if err != nil {
+				return errors.Errorf("Could not access Kubernetes cluster, check kubeconfig: %s", err)
+			}
+			trafficPolicyCheckCmd.clientSet = clientset
+
+			accessCliemt, err := smiAccessClient.NewForConfig(config)
+			if err != nil {
+				return errors.Errorf("Could not initialize SMI Access client: %s", err)
+			}
+			trafficPolicyCheckCmd.smiAccessClient = accessCliemt
+
+			return trafficPolicyCheckCmd.run()
+		},
+		Example: trafficPolicyCheckExample,
+	}
+
+	return cmd
+}
+
+func (cmd *trafficPolicyCheckCmd) run() error {
+	// Validate input for options
+	srcNs, srcPodName, err := unmarshalNamespacedPod(cmd.sourcePod)
+	if err != nil {
+		return errors.Errorf("Invalid argument specified for the source pod: %s", err)
+	}
+
+	dstNs, dstPodName, err := unmarshalNamespacedPod(cmd.destinationPod)
+	if err != nil {
+		return errors.Errorf("Invalid argument specified for the destination pod: %s", err)
+	}
+
+	srcPod, err := cmd.getMeshedPod(srcNs, srcPodName)
+	if err != nil {
+		return err
+	}
+	dstPod, err := cmd.getMeshedPod(dstNs, dstPodName)
+	if err != nil {
+		return err
+	}
+
+	return cmd.checkTrafficPolicy(srcPod, dstPod)
+}
+
+func (cmd *trafficPolicyCheckCmd) checkTrafficPolicy(srcPod, dstPod *corev1.Pod) error {
+	osmNamespace := settings.Namespace()
+
+	// Check if permissive mode is enabled, in which case every meshed pod is allowed to communicate with each other
+	if permissiveMode, err := cmd.isPermissiveModeEnabled(); err != nil {
+		return errors.Errorf("Error checking if permissive mode is enabled")
+	} else if permissiveMode {
+		fmt.Fprintf(cmd.out, "[+] Permissive mode enabled for mesh operated by osm-controller running in %s namespace\n\n "+
+			"[+] Pod '%s/%s' is allowed to communicate to pod '%s/%s'\n",
+			osmNamespace, srcPod.Namespace, srcPod.Name, dstPod.Namespace, dstPod.Name)
+		return nil
+	}
+
+	// SMI traffic policy mode
+	fmt.Fprintf(cmd.out, "[+] SMI traffic policy mode enabled for mesh operated by osm-controller running in %s namespace\n\n", osmNamespace)
+	trafficTargets, err := cmd.smiAccessClient.AccessV1alpha2().TrafficTargets(dstPod.Namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return errors.Errorf("Error listing SMI TrafficTarget policies: %s", err)
+	}
+
+	var foundTrafficTarget bool
+	for _, trafficTarget := range trafficTargets.Items {
+		spec := trafficTarget.Spec
+		// Map traffic targets to the given pods
+		if spec.Destination.Name == dstPod.Spec.ServiceAccountName && spec.Destination.Namespace == dstPod.Namespace {
```

Comment:
It seems to be that both for the destination and source below - we need to also constrain to `spec.Destination.Kind == "ServiceAccount"`

```yaml
kind: TrafficTarget
metadata:
  name: path-specific
  namespace: default
spec:
  destination:
    kind: ServiceAccount
    name: service-a
    namespace: default
    port: 8080
  rules:
  - kind: HTTPRouteGroup
    name: the-routes
    matches:
    - metrics
  sources:
  - kind: ServiceAccount
    name: prometheus
    namespace: default
```

Reply:
Probably needs to happen in the controller as well.